### PR TITLE
MULE-19181: Use the right ClassLoader to handle HTTP requests

### PR DESCRIPTION
**Pull request by eze210**

### CODE REVIEW

File: src/test/java/org/mule/service/http/impl/service/server/grizzly/AbstractGrizzlyServerManagerTestCase.java [hunk 1]
```diff
@@ -311,6 +315,43 @@ public void onlyOwnerCanDisposeServer() throws Exception {
...
+                                      (int) SECONDS.toMillis(DEFAULT_TEST_TIMEOUT_SECS),
+                                      new ServerIdentifier("context", "name"),
+                                      () -> muleContext.getConfiguration().getShutdownTimeout());
+    final ResponseStatusCallback responseStatusCallback = mock(ResponseStatusCallback.class);
+    Reference<ClassLoader> requestHandlerExecutionClassLoader = new Reference<>();
+
+    // The request handler is added using this class loader.
+    ClassLoader requestHandlerAdditionClassLoader = mock(ClassLoader.class);
+    withContextClassLoader(requestHandlerAdditionClassLoader, () -> {
+      server.addRequestHandler("/path", (requestContext, responseCallback) -> {
```
Comment (fsgonz) [inline src/test/java/org/mule/service/http/impl/service/server/grizzly/AbstractGrizzlyServerManagerTestCase.java:335]: I would set the path o a constant.

**Review by fsgonz — CHANGES_REQUESTED** [review]

File: src/main/java/org/mule/service/http/impl/service/server/grizzly/GrizzlyHttpServer.java [hunk 2]
```diff
@@ -45,6 +50,7 @@
 public class GrizzlyHttpServer implements HttpServer, Supplier<ExecutorService> {
 
   protected static final Logger logger = LoggerFactory.getLogger(GrizzlyHttpServer.class);
+  private static boolean REPLACE_CONTEXT_CLASSLOADER = getProperty(MULE_LOG_SEPARATION_DISABLED) == null;
```
[the remark this answers is not in the document]
Comment (eze210) [reply src/main/java/org/mule/service/http/impl/service/server/grizzly/GrizzlyHttpServer.java:53]: The property already exists

File: src/main/java/org/mule/service/http/impl/service/server/grizzly/GrizzlyHttpServer.java [hunk 3]
```diff
@@ -164,20 +170,43 @@ public boolean isStopped() {
...
+        setContextClassLoader(currentThread(), outerClassLoader, innerClassLoader);
+        try {
+          requestHandler.handleRequest(requestContext, responseCallback);
+        } finally {
+          setContextClassLoader(currentThread(), innerClassLoader, outerClassLoader);
+        }
+      }
+
+      @Override
+      public ClassLoader getContextClassLoader() {
```
[the remark this answers is not in the document]
Comment (eze210) [reply src/main/java/org/mule/service/http/impl/service/server/grizzly/GrizzlyHttpServer.java:204]: If it ins't necessary, the setter doesn't do anything

File: src/test/java/org/mule/service/http/impl/service/server/grizzly/AbstractGrizzlyServerManagerTestCase.java (hunk 1 above)
Comment (eze210) [reply src/test/java/org/mule/service/http/impl/service/server/grizzly/AbstractGrizzlyServerManagerTestCase.java]: Done :)

**Review by fsgonz — APPROVED** [review]

**Pull request merged**